pkg/cmd/server/origin/audit.go

```diff
@@ -0,0 +1,133 @@
+package origin
+
+import (
+	"bufio"
+	"errors"
+	"fmt"
+	"io"
+	"net"
+	"net/http"
+	"time"
+
+	"github.com/golang/glog"
+	"github.com/pborman/uuid"
+
+	utilnet "k8s.io/apimachinery/pkg/util/net"
+	apifilters "k8s.io/apiserver/pkg/endpoints/filters"
+	apiresponsewriters "k8s.io/apiserver/pkg/endpoints/handlers/responsewriters"
+	apirequest "k8s.io/apiserver/pkg/endpoints/request"
+)
+
+const AUDIT_TRIGGERED = "audittriggered"
+
+var _ http.ResponseWriter = &auditResponseWriter{}
+
+type auditResponseWriter struct {
```

Comment:
a good comment would be nice. It doesn't sound like this is just the fallback for the upstream audit handler. Maybe call it authFallbackAuditResponseWriter?